pennylane_lightning/lightning_tensor/lightning_tensor.py

```diff
@@ -0,0 +1,337 @@
+            ``np.complex64`` or ``np.complex128``.
+        **kwargs: keyword arguments.
+    """
+
+    # TODO: decide whether to move some of the attributes in interfaces classes
+    # pylint: disable=too-many-instance-attributes
+
+    # So far we just insert the options for MPS simulator
+    _device_options = (
+        "apply_reverse_lightcone",
+        "backend",
+        "c_dtype",
+        "cutoff",
+        "method",
+        "max_bond_dim",
+        "measure_algorithm",
+        "return_tn",
+        "rehearse",
+    )
+
+    _new_API = True
+
+    # TODO: decide if `backend` and `method` should be keyword args as well
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        *,
+        wires=None,
+        backend="quimb",
+        method="mps",
+        shots=None,
+        c_dtype=np.complex128,
+        **kwargs,
+    ):
+
+        if not accepted_backends(backend):
+            raise ValueError(f"Unsupported backend: {backend}")
+
+        if not accepted_methods(method):
+            raise ValueError(f"Unsupported method: {method}")
+
+        if shots is not None:
+            raise ValueError("LightningTensor does not support the `shots` parameter.")
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._num_wires = len(self.wires) if self.wires else 0
+        self._backend = backend
+        self._method = method
+        self._c_dtype = c_dtype
+
+        # options for MPS
+        self._max_bond_dim = kwargs.get("max_bond_dim", None)
+        self._cutoff = kwargs.get("cutoff", 1e-16)
+        self._measure_algorithm = kwargs.get("measure_algorithm", None)
+
+        # common options (MPS and TN)
+        self._apply_reverse_lightcone = kwargs.get("apply_reverse_lightcone", None)
+        self._return_tn = kwargs.get("return_tn", None)
+        self._rehearse = kwargs.get("rehearse", None)
+
+        self._interface = None
+
+        # TODO: implement the remaining combs of `backend` and `interface`
+        if self.backend == "quimb" and self.method == "mps":
+            self._interface = QuimbMPS(self._num_wires, self._c_dtype)
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "lightning.tensor"
+
+    @property
+    def num_wires(self):
+        """Number of wires addressed on this device."""
+        return self._num_wires
+
+    @property
+    def backend(self):
+        """Supported backend."""
+        return self._backend
+
+    @property
+    def method(self):
+        """Supported method."""
+        return self._method
+
+    @property
+    def c_dtype(self):
+        """State vector complex data type."""
+        return self._c_dtype
+
+    dtype = c_dtype
+
+    def _setup_execution_config(self, config):
+        """
+        Update the execution config with choices for how the device should be used and the device options.
+        """
+        updated_values = {}
+        if config.use_device_gradient is None:
```

Comment:
Will these devices support any kind of device derivatives? Im wondering if device derivatives with mps would be a whole other project.

Reply:
Actually, I inserted that line at first, but I had to erase it since Codecov was complaining (tests can't cover it since gradients are out of the scope for now). I can re-insert the line and comment it, but so far I would say it is not crucial